### PR TITLE
Remove latency delay and return real responses

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-648

### CODE REVIEW

File: internal/publish/publish.go [hunk 1]
```diff
@@ -120,9 +117,8 @@ func (h *publishHandler) handleRequest(w http.ResponseWriter, r *http.Request) r
 		// isn't transient.
 		// This message (and logging) will only contain the AppPkgName from the request
 		// and no other data from the request.
-		msg := fmt.Sprintf("no AuthorizedApp, dropping data: %v", err)
-		logger.Error(msg)
-		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: msg})
+		message := fmt.Sprintf("no AuthorizedApp, dropping data: %v", err)
+		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: message})
 		return response{
 			status:      http.StatusInternalServerError,
```
Comment (mikehelmick) [inline internal/publish/publish.go:123]: should this be unauthorized now?

Comment (sethvargo) [reply internal/publish/publish.go]: Done

File: internal/publish/publish.go [hunk 2]
```diff
@@ -120,12 +117,11 @@ func (h *publishHandler) handleRequest(w http.ResponseWriter, r *http.Request) r
...
-		msg := fmt.Sprintf("no AuthorizedApp, dropping data: %v", err)
-		logger.Error(msg)
-		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: msg})
+		message := fmt.Sprintf("no AuthorizedApp, dropping data: %v", err)
+		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: message})
 		return response{
-			status:      http.StatusInternalServerError,
-			message:     http.StatusText(http.StatusInternalServerError),
+			status:      http.StatusUnauthorized,
+			message:     message,
```
Comment (mikehelmick) [inline internal/publish/publish.go:124]: sorry - should have caught this - this should be the standard http.StatusText( as well and not say "NoAuthorizedApp found"

Comment (sethvargo) [reply internal/publish/publish.go:124]: How would the app know why it was unauthorized then? We already return the real message for if the app posts to a bad region, for example.

Comment (mikehelmick) [reply internal/publish/publish.go:124]: true.

**Pull request merged**